### PR TITLE
Start invoice info

### LINKED ISSUE #24: Create Invoice Class
The invoice has the following data accessible:

id - returns the integer id
customer_id - returns the customer id
merchant_id - returns the merchant id
status - returns the status
created_at - returns a Time instance for the date the item was first created
updated_at - returns a Time instance for the date the item was last modified
We create an instance like this:

```rb
i = Invoice.new({
  :id          => 6,
  :customer_id => 7,
  :merchant_id => 8,
  :status      => "pending",
  :created_at  => Time.now,
  :updated_at  => Time.now,
})
```

Effort point 1

### PR DESCRIPTION
**Pull request by sami-p**
Changes implemented: 

- Invoice Class file
- Invoice Spec file
- Invoice Class initialize method with transaction_details info 
- Invoice Spec initialize tests with transaction_details info

closes #24 

- [x] tests passing
- [x] ready to merge

### CODE REVIEW

File: spec/invoice_spec.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        :updated_at  => Time.now,
+      })
+
+      expect(i).is_a? Invoice
+    end
+
+    it 'has an id' do
+      i = Invoice.new({
+        :id          => 6,
+        :customer_id => 7,
```
Comment (NoahZinter) [inline spec/invoice_spec.rb]: I think hound will be cranky about this spacing now. I remember we talked about doing it this way but then we jumped over to whatever rubocop does with it

Comment (sami-p) [reply spec/invoice_spec.rb]: @NoahZinter Thanks for the catch!!

**Review by NoahZinter — APPROVED** [review]
Looks good now!

File: data/mock_data.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 require 'Date'
 
 class MockData
+
+  NO_SELF_ERROR_MESSAGE = 'use {self} at the end of your *_as_mocks method'
```
Comment (mkrumholz) [inline data/mock_data.rb:5]: Nice safety add here.

File: data/mock_data.rb [hunk 3]
```diff
@@ -14,11 +17,62 @@ def self.get_a_random_price
...
+    invoice_hashes.each do |invoice_hash|
+      raise NO_SELF_ERROR_MESSAGE if not block_given?
+      eg = yield
+      invoice_mock = eg.instance_double('Invoice',
+        id: invoice_hash[:id],
+        customer_id: invoice_hash[:customer_id],
+        merchant_id: invoice_hash[:merchant_id],
+        created_at: invoice_hash[:created_at],
+        updated_at: invoice_hash[:updated_at]
+      )
```
Comment (mkrumholz) [inline data/mock_data.rb:42]: This is a note for later, but we may need to consider how we're mocking times at some point. I've got it working well on items for now, though, so may not be necessary.

File: data/mock_data.rb [hunk 4]
```diff
@@ -14,11 +17,62 @@ def self.get_a_random_price
...
+      mocked_invoices << invoice_mock
+    end
+    mocked_invoices
+  end
+
+  def self.invoices_as_hash(number_of_mocks: 10, random_dates: true,
```
Comment (mkrumholz) [inline data/mock_data.rb:48]: Like the amount of customization here. Seems super useful.

File: spec/invoice_spec.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  describe '#initialize' do
+    it 'exists' do
+      i = Invoice.new({
+                        id: 6,
+                        customer_id: 7,
+                        merchant_id: 8,
+                        status: 'pending',
+                        created_at: Time.now,
+                        updated_at: Time.now
+                      })
```
Comment (mkrumholz) [inline spec/invoice_spec.rb:14]: Testing looks thorough here. Nice!

**Review by mkrumholz — APPROVED** [review]
This is looking really great so far. We may need to run the spec harness to see what's needed for time formatting, but I'd say this is ready to merge and we can try that on main as needed.

**Pull request merged**